pkg/runtime/interfaces.go

```diff
@@ -20,29 +20,23 @@ import (
 	"io"
 )
 
-// ObjectScheme represents common conversions between formal external API versions
-// and the internal Go structs. ObjectScheme is typically used with ObjectCodec to
-// transform internal Go structs into serialized versions. There may be many valid
-// ObjectCodecs for each ObjectScheme.
-type ObjectScheme interface {
-	ObjectConvertor
-	ObjectTyper
-	ObjectCreater
-	ObjectCopier
-}
-
-// ObjectCodec represents the common mechanisms for converting to and from a particular
-// binary representation of an object.
-type ObjectCodec interface {
-	ObjectEncoder
+// Codec defines methods for serializing and deserializing API objects.
+type Codec interface {
 	Decoder
+	Encoder
+
+	// Methods for processing url parameters.
+	// EncodeParameters(obj Object) (url.Values, obj Object) error
+	// DecodeParametersInto(parameters url.Values, obj Object) error
```

Comment:
So is `EncodeParameters` only supposed to return `url.Values` and `error`? Also, why can't `DecodeParametersInto` return an object?

As far as where this belongs, I think clients and servers should separately define the interfaces they want to consume in their packages. So the client version of this interface wouldn't even have a decode function.